indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -372,12 +373,10 @@ private TaskStatus runInternal(TaskToolbox toolbox) throws Exception
     // Now we can initialize StreamChunkReader with the given toolbox.
     final StreamChunkParser parser = new StreamChunkParser(
         this.parser,
-        new SettableByteEntityReader(
-            inputFormat,
-            inputRowSchema,
-            task.getDataSchema().getTransformSpec(),
-            toolbox.getIndexingTmpDir()
-        )
+        inputFormat,
```

Comment:
I think it would be good to add some tests for this class since it looks like the existing unit tests don't cover this class at all. For example, there's nothing ensuring that `parser` and `inputFormat` are both not `null` when they're passed to the `StreamChunkParser` constructor.

Also, it looks like we only have tests that ensure that real time ingestion works with JSON. We should add a tests to make sure real time ingestion never breaks in the future for Avro, TSV, etc.

Reply:
> I think it would be good to add some tests for this class since it looks like the existing unit tests don't cover this class at all. For example, there's nothing ensuring that `parser` and `inputFormat` are both not `null` when they're passed to the `StreamChunkParser` constructor.

Hmm, I'm not sure I understood your comment. I believe this class is being tested in both `KafkaIndexTaskTest` and `KinesisIndexTaskTest`. Either `parser` or `inputFormat` can be set for kafka indexing service to work. If both are set, it prefers `inputFormat` over `parser.

> Also, it looks like we only have tests that ensure that real time ingestion works with JSON. We should add a tests to make sure real time ingestion never breaks in the future for Avro, TSV, etc.

Those tests seem useful, but I don't think adding them should be done in this PR since it's out of scope; the issue is creating a `SettableByteEntityReader` can fail with any parser in which `toInputFormat` is missing. I will add them in a follow-up PR.

Reply:
Cool, I didn't look at the coverage from `KafkaIndexTaskTest` and `KinesisIndexTaskTest`, so my comment can be disregarded.

Adding the integration tests in a follow up PR sounds good to me since you manually verified your fix for the scenario that uncovered this bug.